Advanced table design docs (#2647)
hashicorp/design-system: website/docs/components/table/advanced-table/partials/specifications/states.md

## States

> ### Advance Table header columns

Reviewer: The header doesn't have a state, the button has a state.

```suggestion
### Sort button
```

Sortable header columns have interactive state variants, indicating their interactivity. Non-sortable header columns are static and do not include state variants.

![Header column state example](/assets/components/table/advanced-table/Advanced-table-sort-button-states.png)

### Advanced Table cells 

> Cells have a default state and a focused state when active, providing visual feedback during user interaction. Non-interactive columns remain static without state changes.

Reviewer: ```suggestion
Cells have a default and focused state. The focused state provides visual feedback for keyboard navigation. 
```

> ## States

Reviewer 2: There are also the cell focused state and focus states for the cell expand/collapse button.

### Advance Table header columns

> Only sortable header columns have state variants. Non-sortable header columns are not interactive and therefore do not have interactive states.

Reviewer 3: Is this copied over from the table docs? This isn't true for this component. I think some additional content is needed covering the interactive states for the headers/cells themselves, as well as the content within the cell.

Author: Yeah I need to revise this content.
